src/main/java/org/gradle/profiler/CommandExec.java

```diff
@@ -248,10 +254,30 @@ public void interrupt() {
         }
 
         public void kill() {
+            // On Windows subprocesses are not automatically
+            // killed with parent, so let's do that manually
+            destroyDescendants();
             process.destroy();
             shutdownExecutor();
         }
 
+        private void destroyDescendants() {
+            if (!JavaVersion.current().isJava9Compatible()) {
+                // ProcessHandle API is available only from JDK9
+                return;
+            }
+            try {
+                @SuppressWarnings("unchecked")
+                Stream<Object> stream = (Stream<Object>) Process.class.getMethod("descendants").invoke(process);
+                Method destroy = Class.forName("java.lang.ProcessHandle").getMethod("destroy");
+                for (Object child : stream.collect(Collectors.toList())) {
```

Comment:
:x: I think you can call `forEach()` here instead, no?

Reply:
The problem is, that invoke throws exception. But moved to a separate method and used forEach